Source/reactions/Castro_react.cpp

```diff
@@ -97,8 +103,35 @@ Castro::react_state(MultiFab& s, MultiFab& r, Real time, Real dt, const int stra
             Real rhoInv = 1.0_rt / U(i,j,k,URHO);
 
             burn_state.rho = U(i,j,k,URHO);
-            burn_state.T   = U(i,j,k,UTEMP);
-            burn_state.e   = 0.0_rt; // Energy generated by the burn
+            burn_state.T = U(i,j,k,UTEMP);
+
+#ifdef CXX_MODEL_PARSER
+            if (drive_initial_convection) {
+                Real rr[3] = {0.0_rt};
+
+                rr[0] = problo[0] + dx[0] * (static_cast<Real>(i) + 0.5_rt) - problem::center[0];
+#if AMREX_SPACEDIM >= 2
+                rr[1] = problo[1] + dx[1] * (static_cast<Real>(j) + 0.5_rt) - problem::center[1];
+#endif
+#if AMREX_SPACEDIM == 3
+                rr[2] = problo[2] + dx[2] * (static_cast<Real>(k) + 0.5_rt) - problem::center[2];
+#endif
+
+                Real dist;
+
+                if (domain_is_plane_parallel) {
+                    dist = rr[AMREX_SPACEDIM-1];
+                } else {
+                    dist = std::sqrt(rr[0] * rr[0] + rr[1] * rr[1] + rr[2] * rr[2]);
+                }
+
+                burn_state.T = interpolate(dist, model::itemp);
```

Comment:
Did you want T_fixed here?

Reply:
I can set that just to be safe, but T_fixed is not used by Strang